pkg/ipamplugin/ipam_plugin.go

```diff
@@ -285,6 +303,38 @@ func cmdAdd(args *skel.CmdArgs) error {
 	return cnitypes.PrintResult(r, conf.CNIVersion)
 }
 
+type unlockFn func()
+
+// acquireIPAMLockBestEffort attempts to acquire the IPAM file lock, blocking if needed.  If an error occurs
+// (for example permissions or missing directory) then it returns immediately.  Returns a function that unlocks the
+// lock again (or a no-op function if acquiring the lock failed).
+func acquireIPAMLockBestEffort(path string) unlockFn {
+	log.Info("About to acquire host-wide IPAM lock.")
+	if path == "" {
+		path = ipamLockPath
+	}
+	err := os.MkdirAll(filepath.Dir(path), 0777)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to make directory for IPAM lock")
+		// Fall through, still a slight chance the file is there for us to access.
+	}
+	ipamLock := flock.New(path)
+	err = ipamLock.Lock()
```

Comment:
If the lock is held forever (e.g., due to an error releasing the lock in the instance that is holding it) then I think we'll just block IPAM allocations on this node forever. Is that right?

Maybe we need some way to timeout held locks if a particular lock isn't released within a reasonable amount of time?

Reply:
When the process exits, the lock will be released; a dead process can't hold a file lock.  I believe that we time out the CNI plugin with a panic in a background goroutine so that we can't block forever?